sound/soc/intel/boards/sof_cs42l42.c

```diff
@@ -186,37 +178,14 @@ static int sof_card_late_probe(struct snd_soc_card *card)
 {
 	struct sof_card_private *ctx = snd_soc_card_get_drvdata(card);
 	struct snd_soc_component *component = NULL;
-	char jack_name[NAME_SIZE];
 	struct sof_hdmi_pcm *pcm;
-	int err;
 
 	if (list_empty(&ctx->hdmi_pcm_list))
 		return -EINVAL;
 
-	if (ctx->common_hdmi_codec_drv) {
-		pcm = list_first_entry(&ctx->hdmi_pcm_list, struct sof_hdmi_pcm,
-				       head);
-		component = pcm->codec_dai->component;
-		return hda_dsp_hdmi_build_controls(card, component);
-	}
-
-	list_for_each_entry(pcm, &ctx->hdmi_pcm_list, head) {
-		component = pcm->codec_dai->component;
-		snprintf(jack_name, sizeof(jack_name),
-			 "HDMI/DP, pcm=%d Jack", pcm->device);
-		err = snd_soc_card_jack_new(card, jack_name,
-					    SND_JACK_AVOUT, &pcm->hdmi_jack);
-
-		if (err)
-			return err;
-
-		err = hdac_hdmi_jack_init(pcm->codec_dai, pcm->device,
-					  &pcm->hdmi_jack);
-		if (err < 0)
-			return err;
-	}
-
-	return hdac_hdmi_jack_port_init(component, &card->dapm);
+	pcm = list_first_entry(&ctx->hdmi_pcm_list, struct sof_hdmi_pcm, head);
+	component = pcm->codec_dai->component;
+	return hda_dsp_hdmi_build_controls(card, component);
```

Comment:
@brentlu Can you explain more why can we remove hdac-hdmi support? Is mach->mach_params.common_hdmi_codec_drv always true now?

Reply:
According to previous comment, hdac-hdmi is used on GLK Chromebooks which matches my experience. I upgrade one GLK Chromebook to Chrome-v5.4 kernel. It works well with common_hdmi_codec_drv is true. It seems to me we could get rid of hdac-hdmi since SND_SOC_SOF_HDA_AUDIO_CODEC becomes dependency of SOF machine drivers and it causes mach->mach_params.common_hdmi_codec_drv to be true.